meta-mbl-distro/recipes-core/base-files/base-files_%.bbappend

```diff
@@ -42,6 +44,23 @@ python __anonymous() {
         d.appendVar("FSTAB_LINES", fstab_line)
 }
 
+# Write the values of all the variables created by mbl-partitions.bbclass.
+# Each variable value will be in a file with the variable name as the
+# file name. These files will be placed in /config/factory.
+python do_write_partition_vars() {
+    import pathlib
+
+    workdir = pathlib.Path(d.getVar("WORKDIR", True))
+    for var in d.getVar("MBL_PARTITION_VARS", True).split():
+        var_value = d.getVar(var, True)
+        file_path = pathlib.Path(workdir, var)
+        file_path.write_text(var_value)
+        d.appendVar("MBL_PARTITION_FILES_STAGING_DIR", str(file_path))
+}
+
+addtask write_partition_vars before do_install
+do_write_partition_vars[vardeps] += "MBL_PARTITION_VARS"
```

Comment:
In addition to `MBL_PARTITION_VARS`, you should add `${MBL_PARTITION_VARS}` - each variable needs to be added to vardeps. You've also used `MBL_PARTITION_FILES_STAGING_DIR`... TBH, I think it would be easier to write this function in shell and let BitBake find most of the vardeps.

Reply:
I had an issue writing this function in shell, when I tried to use python inline variable expansion it complained about a bad substitution, I had some code similar to:

```
for var in ${MBL_PARTITION_VARS}; do
    printf "%s" "${@ d.getVar('${var}', True)}" > ${PARTITION_VARIABLES_DIR}/${var}
done
```
Seemed like it didn't like the `${var}` expansion in the inline python expression - which is why I ended up with this.

Reply:
Ah, okay.